cylc/flow/parsec/config.py

```diff
@@ -178,7 +182,36 @@ def idump(self, items=None, sparse=False, prefix='',
                 mkeys.append(j)
         if null:
             mkeys = [[]]
-        self.mdump(mkeys, sparse, prefix, oneline, none_str, handle=handle)
+        if json:
+            self.jdump(mkeys, sparse, prefix, oneline, none_str, handle=handle)
+        else:
+            self.mdump(mkeys, sparse, prefix, oneline, none_str, handle=handle)
+
+    def jdump(
+        self,
+        mkeys=None,
+        sparse=False,
+        prefix='',
+        oneline=False,
+        none_str='',
+        handle=None
+    ):
+        """Dump a config to JSON format.
+        """
+        # When we call json.dumps we use indent to
+        # control whether output is multi-line:
+        indent = None
+        if not oneline:
+            indent = 4
+
+        for keys in mkeys:
```

Comment:
Object of type "None" cannot be used as iterable value

Reply:
How did you get this message?

Reply:
VSCode's Python (Pylance/Pyright) type analysis